File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    type_cnt = {}
    answer = 1
    
    for wear, kind in clothes:
        type_cnt[kind] = type_cnt.get(kind, 0) +1
        
    for kind in type_cnt:
        answer *= (type_cnt[kind] + 1 )
    
    return answer -1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    let closet = {};
    
    clothes.forEach(([name, type]) => {
        if (closet[type]) {
            closet[type]++;
        } else {
            closet[type] = 1;
        }
    });
    
    for (const type in closet) {
        answer *= (closet[type] + 1);
    }
    
    return answer - 1;
}
